pkg/etcd/etcd.go

```diff
@@ -554,11 +568,11 @@ func (e *ETCD) join(ctx context.Context, clientAccessInfo *clientaccess.Info) er
 		return err
 	}
 
-	client, err := getClient(clientCtx, e.config, clientURLs...)
+	client, conn, err := getClient(clientCtx, e.config, clientURLs...)
 	if err != nil {
 		return err
 	}
-	defer client.Close()
```

Comment:
the client does not need to close anymore?

Reply:
because we close the connection instead. the client itself no longer wraps the connection and does not have any state that needs to be release by closing it.